### PR TITLE
Public Release

### PR DESCRIPTION
**Pull request by stelabouras**
This PR includes the final commits for the public release of the SDK

### CODE REVIEW

File: Sources/TransifexNative/Logging.swift
Comment (dbendilas) [inline Sources/TransifexNative/Logging.swift]: The way I had this in mind is to make it public, as a fast way for the clients to define a different log level. e.g.

```swift
public class StandardLogHandler : TXLogHandler {

    private var logLevel: TXLogLevel;

    init(_ logLevel: TXLogLevel) {
        self.logLevel = logLevel
    }
    
    func verbose(_ message: String) { log(message, .verbose) }
    func info(_ message: String) { log(message, .info) }
    func warning(_ message: String) { log(message, .warning) }
    func error(_ message: String) { log(message, .error) }

    private func log(_ message: String, logLevel: TXLogLevel) {
        guard logLevel >= self.logLevel else {
            return
        }
        print(message)
    }

}
```

and in the client:

```swift
TXLogger.setHandler(StandardLogHandler(.info))
```

This way, devs won't have to create their only handler class simply in order to set a new log level.

**Review by dbendilas — COMMENTED** [review]
Everything looks good. There is only one thing about `StandardLogHandler`.

File: Sources/TransifexNative/Logging.swift
Comment (stelabouras) [reply Sources/TransifexNative/Logging.swift]: Just pushed the commit with the updated implementation.

File: Sources/TransifexNative/Core.swift
Comment (dbendilas) [inline Sources/TransifexNative/Core.swift:70]: Didn't check, but it looks like this could fit into 1 line.

File: Sources/TransifexNative/Core.swift
Comment (dbendilas) [inline Sources/TransifexNative/Core.swift:29]: It feels like this could/should be even more descriptive. I try to imagine myself bumping into this code 1 year from now and I don't think I'll remember the exact case this refers to. Maybe mention something about **when** this case appears (i.e. when those key=value files are used).

**Review by dbendilas — COMMENTED** [review]
The new changes look great! I've only left a couple of small comments.

File: Sources/TransifexNative/Core.swift
Comment (stelabouras) [reply Sources/TransifexNative/Core.swift]: True, it fits!

File: Sources/TransifexNative/Core.swift
Comment (stelabouras) [reply Sources/TransifexNative/Core.swift:29]: I have updated the docstring there. I have also updated the `README.md` file and pushed the changes so have a look and let me know :)

**Review by dbendilas — DISMISSED** [review]
Looks great! Good job with the README and the docstrings 👍 
Is there anything left before merging? In any case, I think some commits can be squashed.

**Comment by stelabouras** [comment]
> Looks great! Good job with the README and the docstrings 👍
> Is there anything left before merging? In any case, I think some commits can be squashed.

No there's isn't anything left as far as I can see. When you say 'some' commits, do mean ...all of them? 😃

**Comment by dbendilas** [comment]
> No there's isn't anything left as far as I can see. When you say 'some' commits, do mean ...all of them? 😃

There are too many commits, so I won't go there 😄 . I'll leave it up to you to decide how many commits you believe this PR should have (1 or more)!

**Comment by stelabouras** [comment]
@dbendilas Putting PR back into WIP as we need to implement empty string filtering for `.overrideAll` policy and force cache invalidation for `fetchTranslations()`

**Comment by stelabouras** [comment]
@dbendilas Pushed commits that introduce all changes discussed before, during and after today's presentation. Removing WIP

File: Sources/Transifex/Core.swift
Comment (dbendilas) [inline Sources/Transifex/Core.swift:195]: `he` -> `the`

File: Sources/Transifex/Core.swift
Comment (dbendilas) [inline Sources/Transifex/Core.swift:197]: If I remember, and understand, correctly, it's not just that we don't need to but rather that we shouldn't. Before this change, the behavior wasn't correct, correct? 🙂

**Review by dbendilas — COMMENTED** [review]
Looks great, just a few small comments.

File: Sources/Transifex/Core.swift
Comment (stelabouras) [reply Sources/Transifex/Core.swift:197]: Correct, I will adjust the comment there.

File: Sources/Transifex/Core.swift
Comment (stelabouras) [reply Sources/Transifex/Core.swift:195]: Good catch! 👀

File: Sources/TransifexNative/CDSHandler.swift
Comment (stelabouras) [reply Sources/TransifexNative/CDSHandler.swift:93]: True, I will change it to `.info`.

**Comment by stelabouras** [comment]
@dbendilas Push a final commit with the fixes applied + some changes on the README and introducing a CHANGELOG.md file.

**Review by dbendilas — APPROVED** [review]
Looks great

**Pull request merged**
